bot/exts/filters/filtering.py

```diff
@@ -291,6 +304,71 @@ async def filter_eval(self, result: str, msg: Message) -> bool:
+        """Check if a filter triggers an autoban, and process it if so."""
+        # If the filter reason contains `[autoban]`, we want to auto-ban the user
+        if reason and "[autoban]" in reason.lower():
+            # Create a new context, with the author as is the bot, and the channel as #mod-alerts.
+            # This sends the ban confirmation directly under watchlist trigger embed, to inform
+            # mods that the user was auto-banned for the message.
+            context = await self.bot.get_context(message)
+            context.guild = self.bot.get_guild(Guild.id)
+            context.author = context.guild.get_member(self.bot.user.id)
+            context.channel = self.bot.get_channel(Channels.mod_alerts)
+            context.command = self.bot.get_command("tempban")
+
+            await context.invoke(
+                context.command,
+                message.author,
+                arrow.utcnow() + AUTO_BAN_DURATION,
+                reason=AUTO_BAN_REASON
+            )
+
+    async def _process_redirect(self, url: str, message: discord.Message, stats: Stats) -> None:
+        """Unfurl redirect, run the URL filters on it, and check if further punishment is required."""
+        log.trace(f"Processing redirects for {message.id} by {message.author}.")
+        filter_data = "filter_redirects", self.filters["filter_redirects"]
+        result = await services.unfurl_url(url, max_continues=0, use_cache=True)
+        reason = None
+
+        if result is None:
+            log.debug("Failed to unfurl a redirect.")
+            message_override = ". URL could not be unfurled for an unknown reason."
+            ping = True
+
+        elif result.error is None:
+            has_urls = await self._has_urls(result.destination)
+            if has_urls[0]:
+                log.debug("Unfurled redirect, and destination was blacklisted.")
+                message_override = (
+                    ", and tripped a domain filter.\n\n"
+                    f"Destination: `{result.destination}`\n"
+                    f"Redirects: {result.depth}\n\n"
+                    "Tripped "
+                )
+                if has_urls[1] is not None:
+                    reason = has_urls[1]
+                    await self._process_autoban(reason, message)
+                ping = True
+
+            else:
+                log.debug("Unfurled redirect, but destination was not blacklisted.")
+                message_override = ". URL unfurled, but destination was not blacklisted. Tripped "
+                ping = False
+        else:
+            log.debug("Failed to unfurl a redirect.")
+            message_override = (
+                f".\nFailed to unfurl this URL after {result.depth} attempts due to the following error:\n"
+                f"{result.error}\n\n"
+                f"You can try to manually unfurl with `{constants.Bot.prefix}unfurl {url}`.\n"
```

Comment:
You can pass different arguments to that command to get a different result. It may work, it may not. It's a rather unlikely situation to be in in the first place.